test/Sentinel1MetadataTest.jl

```diff
@@ -0,0 +1,236 @@
+"""
+unitests for the sentinel-1 metadata
+
+    test
+        1)MetaDataSentinel1
+        2)loading data 
+        3)the different sub structures.
+    
+"""
+
+include("../src/separateLater/Sentinel1/Metadata/exampleMetadata.jl")
+include("../src/separateLater/Sentinel1/Metadata/Sentinel1Metadata.jl")
+
+using Test
+import Dates
+
+
+#############################################
+########### Getting the .xml file ###########
+#############################################
+
+
+#dataPath = "data"
+# finding first .SAFE product... using that for testing.
+#safeFolder = (joinpath(dataPath, searchDir(dataPath, "SAFE")[1]))
+#getting annotation files in path
+#xmlFiles = getAnnotationPaths(safeFolder) #return vector of annotation files
+# taking first annotation file XML. 
+#const xmlFile = xmlFiles[1]
+
+const xmlFile = "test/testData/s1a-iw3-slc-vv-20220918t074921-20220918t074946-045056-056232-006.xml"
+
+
+
+
+#############################################
+########### test for MetaDataSentinel1 ###########
+#############################################
+
+
+
+function MetaDataSentinel1Test()
+    slcMetadata = MetaDataSentinel1(xmlFile)
+    checkStructures = isdefined(slcMetadata, :header) && isdefined(slcMetadata, :product) && isdefined(slcMetadata, :image) && isdefined(slcMetadata, :swath) && isdefined(slcMetadata, :burst) && isdefined(slcMetadata, :geolocation)
+    if !checkStructures
+        println("Error in MetaDataSentinel1Test")
+    end
+    return checkStructures
+end
+
+
+#############################################
+########### test for loading data ###########
+#############################################
+
+
+
+function readXmlTest()
+    ## Assert
+    isXML = endswith(xmlFile, ".xml")
+    ## Debug
+    if !isXML
+        println("Input is not .xml format")
+        println(isXML)
+    end
+
+
+    # can the file be read
+    if isXML == true
+        ## Act
+        metaDict = getDictofXml(xmlFile)
+        ## Assert
+        readXMLcheck = metaDict != nothing
+        ## Debug
+        if !readXMLcheck
+            println("Can't load XML file. Error in getDictofXml() ")
+            println(readXMLcheck)
+        end
+        return isXML && readXMLcheck
+    else
+        return isXML
+    end
+end
+
+function XmlDataTest()
```

Comment:
This test could probably be removed.

The other tests should fail if the keys are missing